### PR TITLE
fix: invalid scripts warning fixed for undefined scripts

### PR DESCRIPTION
**Pull request by milaninfy**
Fixes https://github.com/npm/cli/issues/7127

### CODE REVIEW

File: lib/normalize.js [hunk 1]
```diff
@@ -264,7 +264,8 @@ const normalize = async (pkg, { strict, steps, root, changes, allowLegacyCase })
         }
       }
     } else {
-      changes?.push(`Removed invalid "scripts"`)
+      // it only makes sense to fix if package.json has scripts field
+      data.scripts && changes?.push(`Removed invalid "scripts"`)
```
Comment (wraithgar) [inline lib/normalize.js:268]: Let's make this an `if` statement

Comment (milaninfy) [reply lib/normalize.js:268]: Sure, I see some other scenarios as well that I'll cover in next changes

Comment (milaninfy) [reply lib/normalize.js]: These changes were not respecting false values like `null`, `0` or `''`. Instead modified top if condition to check if package has `scripts` then only do the updates.

File: lib/normalize.js [hunk 2]
```diff
@@ -251,7 +251,7 @@ const normalize = async (pkg, { strict, steps, root, changes, allowLegacyCase })
 
   // strip "node_modules/.bin" from scripts entries
   // remove invalid scripts entries (non-strings)
-  if (steps.includes('scripts') || steps.includes('scriptpath')) {
+  if ((steps.includes('scripts') || steps.includes('scriptpath')) && data.scripts !== undefined) {
```
Comment (antongolub) [inline lib/normalize.js:254]: Should we keep the `scripts` section if it's empty? 
Maybe align the logic with the `bin`: https://github.com/npm/package-json/blob/main/lib/normalize.js#L76 ?

Comment (wraithgar) [reply lib/normalize.js:254]: Yes we should.  We don't want to change behavior here unless we have to.  This behavior is very old and at this point we should assume that consumers of npm manifests are treating this attribute as always being there even if empty.

File: test/fix.js [hunk 3]
```diff
@@ -197,6 +197,20 @@ for (const [name, testFix] of Object.entries(testMethods)) {
...
+        }
+        const { content } = await testFix(t, testdir)
+        t.notHas(content, 'scripts')
+      })
+      t.test('falsy scripts', async t => {
+        const testdir = {
+          'package.json': pkg({ scripts: 0 }),
+        }
+        const { content } = await testFix(t, testdir)
+        t.strictSame(content.scripts, undefined)
```
Comment (wraithgar) [inline test/fix.js:212]: This is what I'm talking about w/ the change in behavior.  Today npm would do this:

```sh
~/D/n/s/a $ npm pkg get scripts
0
~/D/n/s/a $ npm pkg fix
~/D/n/s/a $ npm pkg get scripts
{}
```

Currently consumers can reasonably assume that the `scripts` attribute in a registry manifest is an object. Even if empty.

Comment (milaninfy) [reply test/fix.js:212]: I might be wrong here but if package.json has `"scripts":0` and we run `npm pkg fix` it just removes the `scripts` from package.json currently in latest
the reason it `npm pkg get scripts` returns `{ }` is the design of get command if it doesn't find any value for that field.

Comment (wraithgar) [reply test/fix.js:212]:
```
$ cat package.json
{
  "name": "a",
  "version": "1.0.0",
  "main": "index.js",
  "keywords": [],
  "author": "Gar <gar+npm@danger.computer>",
  "license": "ISC",
  "description": ""
}
```

Yep! Good catch.  I think I may have gotten that confused w/ the normalization done during publish.

```
    } else {
      changes?.push(`Removed invalid "scripts"`)
      delete data.scripts
    }
```

It's like this in `scriptpath` which is what `prepare` does (the task done during publish).

**Review by wraithgar — APPROVED** [review]

**Pull request merged**